### DISCUSSION TITLE
fitView for top-center position

### DISCUSSION THREAD

**Discussion by dabosin** [xyflow/xyflow#3155] [Q&A]
Is there a way to set the viewport to top center of the graph? When there are too many nodes and fitView is applied - top and bottom nodes are getting cut off on the initial view.

![IMAGE 2023-06-19 19:22:27](https://github.com/wbkd/react-flow/assets/12236995/8b3aa29a-34e2-42c2-8586-b492ea7bee33)

I know there is a [setViewport](https://reactflow.dev/docs/api/react-flow-instance/#setviewport) and in [this example](https://reactflow.dev/docs/examples/interaction/zoom-transitions/) it kinda does what is needed, but if you open it in [Codesandbox](https://codesandbox.io/s/men5b9?file=%2FApp.js&utm_medium=sandpack) and play with the window - nodes are not centrally aligned with larger chart width.

I know that this can be fixed by providing correct x offset, but not sure how to calculate it since nodes area width depends on parent-child relations of the nodes

**Comment by vasanthid** [comment]
@dabosin - did you find any solution? If so, can you please share?

**Comment by moklick** [comment]
If you want all nodes to fit inside the viewport you need to decrease minZoom (`minZoom={0.1}` for example). Otherwise you need to come up with a certain rule that defines which nodes should be used for calculating the viewport. You could say all nodes with a y < 1000 for example:

```js 
const visibleNodes = nodes.filter(node => node.position.y < 1000);
const nodesRect = getRectOfNodes(visibleNodes);
const transform = getTransformForBounds(nodesRect);
```

https://reactflow.dev/docs/api/graph-util-functions/#getrectofnodes
